uqcsbot/join.py

```diff
@@ -0,0 +1,103 @@
+import discord
```

Comment:
Is creating a separate cog ideal for this? We do already have the channels cog, which I think these should be a part of. Also should help reduce a bit of code duplication

Reply:
My thoughts here were that given ``!joinchannel`` and ``!leavechannel`` are going to be phased out eventually, it would be easier just to separate them from this new feature. However it is definitely ideal to have all the joining/leaving centralised to one cog for the time being so I will merge them in the next commit